develop-docs/sdk/telemetry/logs.mdx

```diff
@@ -0,0 +1,381 @@
+| 21-24                | Fatal      | A fatal error such as application or system crash.                                      |
+
+Default SDK public API should set the lowest severity number for a given severity level. For example, `warn` severity level logs collected by the SDK API should use the severity number `13`.
+
+## Public API
+
+API wise the SDKs are required to expose logging methods which are to be defined in a `logger` module or namespace. The SDKs should also include some initialization options to configure the behavior of logs in the SDK.
+
+### Initialization Options
+
+The SDKs need to expose the following two configuration options:
+
+- `enableSentryLogs`/`enable_sentry_logs`: A boolean flag to control if log envelopes will be generated and sent to Sentry via the Sentry SDK's Logging APIs. If this flag is set to `false`, the SDK should not send logs to Sentry. Defaults to `false`.
+
+- `beforeSendLog`/`before_send_log`: A function that takes a log object and returns a log object. This function is called before sending the log to Sentry. It can be used to modify the log object or to prevent the log from being sent to Sentry.
+
+```js
+Sentry.init({
+  enableSentryLogs: true,
+
+  beforeSendLog(log) {
+    // Prevent logs from being sent to Sentry if the plan type is enterprise
+    if (log.attributes["plan.type"] === "enterprise") {
+      return null;
+    }
+
+    return log;
+  },
+});
+```
+
+At the current moment, logs are not sampled so no sampling configuration needs to be exposed. This may change in the future.
+
+### Logger Module
+
+At minimum the SDK needs to implement the following two items to be considered to have a complete logging public API:
+
+1. `Sentry.logger.X` (where `X` is the log level): The log levels are `trace`, `debug`, `info`, `warn`, `error`, and `fatal`, which is specced out by the [protocol below](#log-severity-level).
+
+- `Sentry.logger.trace`
+- `Sentry.logger.debug`
+- `Sentry.logger.info`
+- `Sentry.logger.warn`
+- `Sentry.logger.error`
+- `Sentry.logger.fatal`
+
+These methods accepts a string template and the parameters to that string template so the SDK can perform structured logging. Optionally these methods take arbitrary attributes, but not all languages can support both passing a parameterized template and attributes in an easy way.
+
+```js
+// Need to use `fmt` helper function in JavaScript for structured logging.
+Sentry.logger.info(Sentry.logger.fmt`Adding item ${itemId} for user ${userId}`);
+
+Sentry.logger.warn("User %s performed action %s", [userId, actionName], {
+  extra: "123",
+});
+```
+
+Optionally the SDK can define a `log` log severity level, which maps to the `info` log severity level in Sentry. If an SDK collects `log` severity level, it should map to `info` severity level and `9` severity number.
+
+```js
+Sentry.logger.log("Added item to shopping cart");
+```
```

Comment:
I'm a bit confused as to why we need this. What's the motivation to have an extra `log` method that is basically `info`?

Reply:
I think I'd be in favour of renaming `emit` to `log` actually, but I understand we're following otel here

Reply:
> I'm a bit confused as to why we need this. What's the motivation to have an extra log method that is basically info?

Just as an alias. Let me actually remove this, I think SDKs including this is a bad idea actually. Done with 5ea9d7d6acdb020939e3a751a2f278c1be30cd9d